bbb-common-web/src/main/java/org/bigbluebutton/presentation/imp/Office2PdfPageConverter.java

```diff
@@ -58,14 +49,33 @@ public static boolean convert(File presentationFile, File output, int page, Uplo
       String logStr = gson.toJson(logData);
       log.info(" --analytics-- data={}", logStr);
 
-      final DocumentFormat sourceFormat = DefaultDocumentFormatRegistry.getFormatByExtension(
-              FilenameUtils.getExtension(presentationFile.getName()));
+      try {
+        log.info("Calling conversion script " + presOfficeConversionExec);
 
-      inputStream = new FileInputStream(presentationFile);
-      outputStream = new FileOutputStream(output);
+        if(presOfficeConversionExec == null) throw new Exception("Cannot find the conversion script path.");
 
-      converter.convert(inputStream).as(sourceFormat).to(outputStream).as(DefaultDocumentFormatRegistry.PDF).execute();
-      outputStream.flush();
+        File conversionScript = new File(presOfficeConversionExec);
+        if(!conversionScript.exists()) throw new Exception(presOfficeConversionExec + ", file not found.");
+
+        Process p = Runtime.getRuntime().exec(String.format(presOfficeConversionExec + " %s %s", presentationFile.getAbsolutePath(), output.getAbsolutePath()));

```

Comment:
use `ProcessBuilder` instead of Runtime.exec ( to avoid shell script injection ): https://docs.oracle.com/javase/6/docs/api/java/lang/ProcessBuilder.html

Reply:
Done!